activerecord/lib/active_record/persistence.rb

```diff
@@ -364,7 +364,10 @@ def destroy
     # and #destroy! raises ActiveRecord::RecordNotDestroyed.
     # See ActiveRecord::Callbacks for further details.
     def destroy!
-      destroy || _raise_record_not_destroyed
+      result = destroy
+      return result if result
+      raise RecordNotDestroyed.new(self.errors.first.second, self) if self.errors.present?
```

Comment:
`self.errors.first.second` feels both fragile and hard to understand what it's supposed to be without executing the code